lib/temporal/worker.rb

```diff
@@ -0,0 +1,72 @@
+require 'async'
+require 'temporal/bridge'
+require 'temporal/data_converter'
+require 'temporal/failure_converter'
+require 'temporal/payload_converter'
+require 'temporal/runtime'
+require 'temporal/worker/activity'
+require 'temporal/worker/thread_pool_executor'
+
+module Temporal
+  class Worker
+    # TODO: Add worker interceptors
+    def initialize(
+      connection,
+      namespace,
+      task_queue,
+      payload_converter: Temporal::PayloadConverter::DEFAULT,
+      payload_codecs: [],
+      failure_converter: Temporal::FailureConverter::DEFAULT,
+      activity_executor: nil,
+      max_concurrent_activities: 100
+    )
+      @running = false
+      @mutex = Mutex.new
+      @runtime = Temporal::Runtime.instance
```

Comment:
Don't have to now, but might want to consider accepting a form of this from the user (it'd usually come from the client, but I know you're not dev'ing that way). We have recently changed to this model in Python because SDK Core now supports multiple runtimes with different telemetry options a user can configure.

If you make it user visible, you can also remove the `reactor` parameter from `start` since the user could alter it on the runtime themselves.